File: DinnerTask.java
1.

public class EvenNumbers {
    public static void main(String[] args) {
        for (int i = 2; i <= 100; i += 2) {
            System.out.print(i + " ");
        }
    }
}

2.

public class OddNumbers {
    public static void main(String[] args) {
        for (int i = 51; i < 100; i += 2) {
            System.out.println(i);
        }
    }
}

3.

public class Countdown {
    public static void main(String[] args) {
        for (int i = 100; i >= 1; i--) {
            System.out.println(i);
        }
    }
}

4.

public class SquareNumbers {
    public static void main(String[] args) {
        for (int i = 1; i <= 20; i++) {
            System.out.println("Square of " + i + " is " + (i * i));
        }
    }
}

5.

public class MultiplesOfThree {
    public static void main(String[] args) {
        for (int i = 1; i <= 50; i++) {
            if (i % 3 == 0) {
                System.out.println(i);
            }
        }
    }
}

6.

public class DivisibleBy3And5 {
    public static void main(String[] args) {
        for (int i = 1; i <= 100; i++) {
            if (i % 3 == 0 && i % 5 == 0) {
                System.out.println(i);
            }
        }
    }
}

7.

public class CountDivisibleBy7 {
    public static void main(String[] args) {
        int count = 0;
        for (int i = 1; i <= 100; i++) {
            if (i % 7 == 0) {
                count++;
            }
        }
        System.out.println("Count of numbers divisible by 7 between 1 and 100: " + count);
    }
}

8.

public class SumFirst50Numbers {
    public static void main(String[] args) {
        int sum = 0;
        for (int i = 1; i <= 50; i++) {
            sum += i;
        }
        System.out.println("Sum of the first 50 natural numbers: " + sum);
    }
}

9.

public class ProductFirst10Numbers {
    public static void main(String[] args) {
        long product = 1;
        for (int i = 1; i <= 10; i++) {
            product *= i;
        }
        System.out.println("Product of the first 10 natural numbers: " + product);
    }
}

10.

public class PrintLetters {
    public static void main(String[] args) {
        for (char c = 'A'; c <= 'Z'; c++) {
            System.out.print(c + " ");
        }
        System.out.println();
    }
}

11.

public class MultiplicationTable {
    public static void main(String[] args) {
        int number = 7;  // Change this number to print a different table
        for (int i = 1; i <= 10; i++) {
            System.out.println(number + " x " + i + " = " + (number * i));
        }
    }
}

12.

public class PrintCharacters {
    public static void main(String[] args) {
        String str = "Hello, World!";
        for (int i = 0; i < str.length(); i++) {
            System.out.println(str.charAt(i));
        }
    }
}

13.

public class CountE {
    public static void main(String[] args) {
        String input = "Here is an example sentence.";
        int count = 0;
        
        for (int i = 0; i < input.length(); i++) {
            if (input.charAt(i) == 'e') {
                count++;
            }
        }
        
        System.out.println("Number of 'e's: " + count);
    }
}

14.

public class LowerToUpper {
    public static void main(String[] args) {
        String input = "Hello World!";
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            if (c >= 'a' && c <= 'z') {
                // Convert to uppercase by subtracting 32 from ASCII value
                c = (char) (c - 32);
            }
            result.append(c);
        }

        System.out.println(result.toString());
    }
}

15.

public class ToLowerManual {
    public static void main(String[] args) {
        String input = "HeLLo WoRLD!";
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            // Check if uppercase A-Z
            if (c >= 'A' && c <= 'Z') {
                // Convert to lowercase by adding 32
                result.append((char)(c + 32));
            } else {
                result.append(c);
            }
        }

        System.out.println("Original: " + input);
        System.out.println("Lowercase: " + result.toString());
    }
}

16.

public class VowelCounter {
    public static void main(String[] args) {
        String input = "Hello World!";
        int count = 0;
        String vowels = "aeiouAEIOU";

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            if (vowels.indexOf(c) != -1) {
                count++;
            }
        }

        System.out.println("Original: " + input);
        System.out.println("Number of vowels: " + count);
    }
}

17.

public class DigitCounter {
    public static void main(String[] args) {
        int number = 123456;
        int count = 0;
        int temp = Math.abs(number); // Handle negative numbers

        if (temp == 0) {
            count = 1;
        } else {
            while (temp > 0) {
                temp /= 10;
                count++;
            }
        }

        System.out.println("Number: " + number);
        System.out.println("Digit count: " + count);
    }
}

18.

public class DigitSum {
    public static void main(String[] args) {
        int number = 12345;
        int sum = 0;

        while (number != 0) {
            sum += number % 10;
            number /= 10;
        }

        System.out.println("Sum of digits: " + sum);
    }
}

19.

public class LargestDigit {
    public static void main(String[] args) {
        int number = 739245;
        int maxDigit = 0;

        while (number > 0) {
            int digit = number % 10;
            if (digit > maxDigit) {
                maxDigit = digit;
            }
            number /= 10;
        }

        System.out.println("Largest digit: " + maxDigit);
    }
}

20.

public class SmallestDigit {
    public static void main(String[] args) {
        int number = 739245;
        int minDigit = 9; // max digit possible

        while (number > 0) {
            int digit = number % 10;
            if (digit < minDigit) {
                minDigit = digit;
            }
            number /= 10;
        }

        System.out.println("Smallest digit: " + minDigit);
    }
}

File: Dinner_task.py
1.

for i in range(2, 101, 2):
    print(i, end=' ')

2.

for i in range(51, 100, 2):
    print(i)

3.

for i in range(100, 0, -1):
    print(i)

4.

for i in range(1, 21):
    print(f"Square of {i} is {i * i}")

5.

for i in range(1, 51):
    if i % 3 == 0:
        print(i)

6.

for i in range(1, 101):
    if i % 3 == 0 and i % 5 == 0:
        print(i)
7.

count = 0
for i in range(1, 101):
    if i % 7 == 0:
        count += 1

print(f"Count of numbers divisible by 7 between 1 and 100: {count}")

8.

sum = 0
for i in range(1, 51):
    sum += i

print(f"Sum of the first 50 natural numbers: {sum}")

9.

product = 1
for i in range(1, 11):
    product *= i

print(f"Product of the first 10 natural numbers: {product}")

10.

for c in range(ord('A'), ord('Z') + 1):
    print(chr(c), end=" ")
print()

11.

number = 7  # Change this number to print a different table
for i in range(1, 11):
    print(f"{number} x {i} = {number * i}")

12.

str = "Hello, World!"
for char in str:
    print(char)

13.

input_str = "Here is an example sentence."
count = input_str.count('e')
print("Number of 'e's:", count)

14.

input_str = "Hello World!"
result = ""

for char in input_str:
    if 'a' <= char <= 'z':
        # Convert to uppercase by subtracting 32 from ASCII code
        result += chr(ord(char) - 32)
    else:
        result += char

print(result)

15.

input_str = "HeLLo WoRLD!"
result = ""

for char in input_str:
    if 'A' <= char <= 'Z':
        # Convert to lowercase using ASCII value
        result += chr(ord(char) + 32)
    else:
        result += char

print("Original:", input_str)
print("Lowercase:", result)

16.

input_str = "Hello World!"
vowels = "aeiouAEIOU"
count = 0

for char in input_str:
    if char in vowels:
        count += 1

print("Original:", input_str)
print("Number of vowels:", count)

17.

number = 123456
count = 0
temp = abs(number)  # Handle negative numbers

if temp == 0:
    count = 1
else:
    while temp > 0:
        temp //= 10
        count += 1

print("Number:", number)
print("Digit count:", count)

18.

number = 12345
sum_of_digits = sum(int(digit) for digit in str(number))
print("Sum of digits:", sum_of_digits)

19.

umber = 739245
largest_digit = max(int(digit) for digit in str(number))
print("Largest digit:", largest_digit)

20.

number = 739245
smallest_digit = min(int(digit) for digit in str(number))
print("Smallest digit:", smallest_digit)


